sqlmesh/core/model/cache.py

```diff
@@ -84,22 +85,27 @@ def with_optimized_query(self, model: Model) -> bool:
         if not isinstance(model, SqlModel):
             return True
 
-        entry_id = self._entry_id(model)
+        unoptimized_query = model.render_query(optimize=False)
+        if unoptimized_query is None:
+            return False
```

Comment:
i think you want return True here so that it doesn't go into validation, that's how i use this return value

Reply:
It won't go into validation because I check whether the query is not none there. I think we shouldn't use this flag to bypass validation. It should instead indicate whether there was a cache hit.

Reply:
then you should fix the previous one that returns True when it shouldn't